### PR TITLE
Fix start with bottles in plentiful item pool

### PR DESCRIPTION
**Pull request by flagrama**
Starting with 3 bottles in plentiful would cause an exception due to not seeing a third bottle to remove as Bottle with Letter does not count as a normal bottle. If not starting with the Bottle with Letter the count of bottles started with is reduced by 1 to compensate. This is done so that there are still 2 Bottle with Letter placed in the world as is expected of the plentiful item pool. This may confuse players who wish to start with 3 bottles, however I believe this behavior is more desirable than an exception, or obtaining a 5th bottle.

Starting with Ruto's Letter in plentiful would start with 1 Bottle with Letter and place one in the world. Starting with Ruto's Letter will now start with letter and place a random bottle in place of the second one.

---
I did repeat myself a bit in this code.  I wasn't sure the best way to check the conditions and make the changes as they can happen in either order due to the starting items being checked in order they were added to the starting items list.

I also haven't checked that you start with the proper number of bottles yet. I think it should work properly due to collect_starters being after alter_pool, but haven't actually tested this in-game yet.

Some spoiler logs I made though:
- [Start with all 4, letter at top](https://pastebin.com/raw/HARr0n8Y)
- [Start with all 4, letter at bottom](https://pastebin.com/raw/LAPvxFd5)
- [Start with 3 bottles, reduced to start with 2 bottles and place 2 bottle with letter](https://pastebin.com/raw/tW65UQ6t)
- [Start with Bottle with Letter](https://pastebin.com/raw/bTksiyMu)

### CODE REVIEW

**Comment by flagrama** [comment]
Added unittests _after_ fixing the bug because I'm a scrub and forgot. The .sav files are useful for ensuring the generation doesn't throw an exception, but I think I needed to use .json files to test the finer details. Could just be me using my plando tests as a hammer and the things that needed tested looked like nails.

**Comment by flagrama** [comment]
That should be better.

File: Plandomizer.py [hunk 1]
```diff
@@ -408,19 +408,32 @@ def alter_pool(self, world, pool):
...
+                        and record.count > 2):
+                    record.count -= 1
+                bottles = record.count
             elif trade_matcher(item_name):
                 self.pool_remove_item([pool], "#AdultTrade", record.count)
             elif IsItem(item_name):
                 try:
-                    self.pool_remove_item([pool], item_name, record.count)
+                    if item_name == "Bottle with Letter":
+                        self.pool_remove_item([pool], item_name, 2)
```
Comment (Zannick) [inline Plandomizer.py:425]: Removing 2 Letters from the pool... is that guaranteed to work? And does it matter how many Bottles with Letter the player asked to start with?

File: Unittest.py [hunk 2]
```diff
@@ -274,6 +274,44 @@ def test_pool_accuracy(self):
             for item in distribution_file['starting_items']:
                 self.assertNotIn(item, actual_pool)
 
+    def test_rutoletter_startwith(self):
+        filenames = [
+            "rutoletter-startwith",
+            "rutoletter-startwithother",
+            "rutoletter-startwithall",
+            "rutoletter-startwithall2"
```
Comment (Zannick) [inline Unittest.py:282]: I think it'd be worthwhile to add a non-plentiful version of one or two of these.

**Pull request closed without merging**